src/lib/components/ui/input/date-input.svelte

```diff
@@ -126,12 +129,23 @@
 		return new Date(year, month - 1, day)
 	}
 
+	function formatDate(date: Date) {
+		if (!(date instanceof Date) || isNaN(date.getTime())) {
+			return ''
+		}
+		const year = date.getFullYear()
+		const month = String(date.getMonth() + 1).padStart(2, '0')
+		const day = String(date.getDate()).padStart(2, '0')
+		return `${year}-${month}-${day}`
+	}
+
 	function inputChange(event: Event) {
 		const input = event.target as HTMLInputElement
 		const newDate = parseDate(input.value)
 		selectedDate = newDate
 		selectedMonth = newDate.getMonth()
 		selectedYear = newDate.getFullYear()
```

Comment:
BTW I think we can simplify here. These values could be `$derived` from `value` instead of using `$state`. Also the `currentDate` is unnecessary and we can use `value` directly.